micrometer-core/src/main/java/io/micrometer/core/instrument/push/PushMeterRegistry.java

```diff
@@ -121,4 +126,14 @@ public void close() {
         super.close();
     }
 
+    // VisibleForTesting
+    long calculateInitialDelay() {
+        long stepMillis = config.step().toMillis();
+        Random random = new Random();
+        long randomOffsetWithinStep = (long) (stepMillis * random.nextDouble()
+                * PERCENT_RANGE_OF_RANDOM_PUBLISHING_OFFSET);
+        long offsetToStartOfNextStep = stepMillis - clock.wallTime() % stepMillis;
+        return offsetToStartOfNextStep + 2 + randomOffsetWithinStep;
```

Comment:
Can we add a comment that says something like:
Wait till the next step + a little more randomly somewhere between 0 and 80% of the step interval + 2ms?

Btw why 2ms?

Reply:
> why 2ms? Because the rollover is +1?

Yes.

Reply:
I added a polish commit with a comment explaining. I also changed the implementation slightly so that the 2ms isn't increasing the latest time a publish can be scheduled. It shouldn't matter for anything but very small steps (like the test case is using).

Reply:
nit: jitter might be a good variable name for the random value.